File: ps/src/main/kotlin/boj/boj_1141/java/Main.java
package boj.boj_1141.java;

import static java.util.Comparator.comparingInt;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        List<String> words = new ArrayList<>();

        for (int index = 0; index < n; index++) {
            words.add(br.readLine());
        }

        words.sort(comparingInt(String::length));
        int count = 0;
        outer:for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (words.get(j).startsWith(words.get(i))) {
                    continue outer;
                }
            }
            count++;
        }
        System.out.println(count);
    }
}

File: ps/src/main/kotlin/boj/boj_1141/kotlin/Main.kt
package boj.boj_1141.kotlin

fun main() {
    val n = readln().toInt()
    val words = mutableListOf<String>()
    for (input in 0 until n) {
        words.add(readln())
    }

    words.sortWith(compareBy<String> { it.length }.thenBy { it })

    val result = mutableListOf<String>()
    outer@ for (i in 0 until words.size) {
        for (j in i + 1 until words.size) {
            if (words[j].startsWith(words[i])) {
                continue@outer
            }
        }
        result.add(words[i])
    }

    println(result.size)
}
